File: pl/java/sort_with_lamda/SortWithLambda.java
import java.util.*;

public class SortWithLambda {
    public static void main(String[] args) {
        List<String> cities = new ArrayList<>(List.of("arta", "ioannina", "preveza", "Igoumenitsa"));
        cities.sort((a, b) -> a.length() - b.length());
        for (String city : cities) {
            System.out.println(city);
        }
    }
}

File: pl/cpp/sort_with_lambda.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    vector<string> cities{"arta", "ioannina", "preveza", "Igoumenitsa"};
    sort(cities.begin(), cities.end(), [](auto a, auto b)
         { return a.length() < b.length(); });
    for (auto city : cities)
    {
        cout << city << endl;
    }
}

// arta
// preveza
// ioannina
// Igoumenitsa